Update from Intelsdi-x pluginsync 0.1.9

nanliu opened the pull request.

rji commented:
👎 this removes required functionality, such as compiling protobuf from the Mesos project, ignoring packages (e.g., `mesos_pb2`) when testing, and removes documentation on how to use the included Vagrant environment, among other things.

rji commented on .travis.yml:
```diff
@@ -1,38 +1,73 @@
+  - TEST_TYPE=medium MESOS_RELEASE=0.28.1-2.0.20.ubuntu1404
+  - TEST_TYPE=medium MESOS_RELEASE=0.27.1-2.0.20.ubuntu1404
+  - TEST_TYPE=medium MESOS_RELEASE=0.26.1-2.0.20.ubuntu1404
+  - TEST_TYPE=build
```
Two of these entries aren't valid version strings. Travis CI build failed.

nanliu replied:
This is updated.

rji commented on CONTRIBUTING.md:
```diff
@@ -1,92 +1,46 @@
-
-If you have questions feel free to contact the [snap maintainers][snap-maintainers].
-
-### Vagrant development environment
```
Seems to me this section would be useful for anyone attempting to develop the plugin in the included Vagrant environment, no?

nanliu replied:
I migrated this to a separate README, since we don't universally have vagrant configs in most of our plugins. Let me know if that's sufficient.

rji commented on LICENSE:
```diff
@@ -175,3 +175,28 @@
 
    END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Not sure why pluginsync has re-added the appendix here. It's not part of the terms and conditions of the Apache license.

nanliu replied:
Added PR to change the template:
https://github.com/intelsdi-x/snap-pluginsync/pull/20

rji commented on Makefile:
```diff
@@ -1,23 +1,37 @@
-	protoc --go_out=import_path=mesos_pb2:mesos/mesos_pb2 mesos_pb2.proto
-	mv mesos/mesos_pb2/mesos_pb2.pb.go mesos/mesos_pb2/mesos_pb2.go
-	rm -f mesos_pb2.proto
+	bash -c "./scripts/build.sh $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))"
```
I ran into issues with this when developing in Vagrant when the working directory was a symlink to the mounted HGFS share, which is why I switched to using `$(PWD)`

nanliu replied:
The new build script should not need the path and can be simplified to: "./scripts/build.sh"

I'll do some test in vagrant before sending a PR to pluginsync.

rji commented on Makefile:
```diff
@@ -1,23 +1,37 @@
 	$(MAKE) test
 all:
-	bash -c "./scripts/build.sh $(PWD)"
-protobuf:
```
This target is required for downloading protobuf from the Mesos project and compiling it to go code which is used throughout this plugin.

rji commented on scripts/test.sh:
```diff
@@ -1,106 +1,100 @@
-COVERALLS_MAX_ATTEMPTS=5
-TEST_DIRS="main.go mesos/"
-PKG_DIRS=". ./mesos/..."
-IGNORE_PKGS="mesos_pb2"
```
We don't want to test the `mesos_pb2` package created by `make protobuf`; it's generated code.

nanliu commented:
@rji, thanks for the review. The changes are mostly applied by a templating tool to standardize all the plugins. This repo has some uniq functionalities and we will make sure it's preserved before applying this update.

nanliu commented on .travis.yml:
```diff
@@ -1,38 +1,73 @@
-    # For more info, see: http://mesos.apache.org/documentation/latest/versioning/
-    #
-    # Fortunately, Mesosphere doesn't remove previous releases from its Apt repositories.
-    # Determine the version string to use here by running `apt-get update && apt-cache policy mesos`
```
This comment is now preserved in .sync.yml config which generates .travis.yml

rji commented:
Awesome, thanks @nanliu! I'm traveling today but I'll try to check that out sometime this weekend.

nanliu commented:
No rush for review, this still needs a bit more touch up, and probably won't be ready until next week with the release of pluginsync 0.1.7.

nanliu commented:
Hmm, I tried updating this to the latest pluginsync version and something has gone wrong with the integration test. The mesos slave agent does not appear to stay in a running state: https://travis-ci.org/intelsdi-x/snap-plugin-collector-mesos/jobs/181762165#L743-L751

```
--------------------------------------------------
Mesos version 0.27.2-2.0.15.ubuntu1404 has been installed.
  * Master: http://127.0.0.1:5050
  * Agent: http://127.0.0.1:5051
--------------------------------------------------
mesos-master start/running, process 5027
mesos-slave stop/waiting
```

PR #41 confirms there's an issue that we need to track down.

nanliu commented:
This PR has been updated to the latest snap plugin sync. I've temporarily dropped 0.27, 0.26 because the mesos-slave service is failing due to:
```
Dec  8 01:27:37 testing-gce-32f27739-cc8c-44ad-b45a-8fd02f9adeb5 mesos-slave[5425]: E1208 01:27:37.842380  5425 perf.cpp:294] Failed to get perf version: Invalid version component '30
Dec  8 01:27:37 testing-gce-32f27739-cc8c-44ad-b45a-8fd02f9adeb5 mesos-slave[5425]: ': Failed to convert '30
Dec  8 01:27:37 testing-gce-32f27739-cc8c-44ad-b45a-8fd02f9adeb5 mesos-slave[5425]: ' to number
Dec  8 01:27:37 testing-gce-32f27739-cc8c-44ad-b45a-8fd02f9adeb5 mesos-slave[5425]: Failed to create a containerizer: Could not create MesosContainerizer: Could not create isolator 'cgroups/perf_event': Perf is not supported
```

Let me know if this addressed your concerns. Once this is merged, we can release binaries with a git tag.

rji commented:
💯 👍 this looks absolutely fantastic, @nanliu! Thanks for putting in the work to get pluginsync working for this plugin.

🙇
